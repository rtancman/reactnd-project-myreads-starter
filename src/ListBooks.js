import React, { Component } from 'react';
import PropTypes from 'prop-types'
import Book from './Book'


class ListBooks extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    books: PropTypes.array.isRequired,
    onMoveShelf: PropTypes.func.isRequired
  }

  render() {
    const { books, title, onMoveShelf} = this.props
    return (
      <div className="bookshelf">
        <h2 className="bookshelf-title">{title}</h2>
        <div className="bookshelf-books">
          <ol className="books-grid">
           {books.map((book) => (
              <li key={book.id}>
                <Book
                  book={book}
                  onMoveShelf={onMoveShelf}
                />
              </li>
            ))}
          </ol>
        </div>
      </div>
    )
  }
}

export default ListBooks

