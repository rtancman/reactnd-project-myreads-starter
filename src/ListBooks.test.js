import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme';
import ListBooks from './ListBooks'
import renderer from 'react-test-renderer';

const books = [
  {
    'title': 'The Linux Command Line',
    'subtitle': 'A Complete Introduction',
    'authors': [
        'William E. Shotts, Jr.'
    ],
    'imageLinks': {
        'smallThumbnail': 'http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api',
        'thumbnail': 'http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api'
    },
    'id': 'nggnmAEACAAJ',
    'shelf': 'currentlyReading'
  },
  {
    'title': 'The Linux',
    'subtitle': 'Introduction',
    'authors': [
        'Shotts Jr.'
    ],
    'id': 'abcenggnmAEACAAJ',
    'shelf': 'currentlyReading'
  }
]

const onMoveShelf = jest.fn()

describe('ListBooks', () => {
  describe('when component renders', () => {
    it('is valid list books', () => {
      const component = renderer.create(
        <ListBooks
          title='Test'
          books={books}
          onMoveShelf={onMoveShelf}
        />
      )
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    })

    it('is empty list books', () => {
      const component = renderer.create(
        <ListBooks
          title='Test'
          books={[]}
          onMoveShelf={onMoveShelf}
        />
      )
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    })

    it('show write title', () => {
      const wrapper = shallow(
        <ListBooks
          title='Test'
          books={[]}
          onMoveShelf={onMoveShelf}
        />
      )
      expect(wrapper.find('h2').text()).toEqual('Test');
    })
  })
})
