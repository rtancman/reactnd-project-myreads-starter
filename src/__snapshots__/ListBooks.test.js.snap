// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBooks when component renders is empty list books 1`] = `
<div
  className="bookshelf"
>
  <h2
    className="bookshelf-title"
  >
    Test
  </h2>
  <div
    className="bookshelf-books"
  >
    <ol
      className="books-grid"
    />
  </div>
</div>
`;

exports[`ListBooks when component renders is valid list books 1`] = `
<div
  className="bookshelf"
>
  <h2
    className="bookshelf-title"
  >
    Test
  </h2>
  <div
    className="bookshelf-books"
  >
    <ol
      className="books-grid"
    >
      <li>
        <div
          className="book"
        >
          <div
            className="book-top"
          >
            <div
              className="book-cover"
              style={
                Object {
                  "backgroundImage": "url(\\"http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api\\")",
                  "height": 193,
                  "width": 128,
                }
              }
            />
            <div
              className="book-shelf-changer"
            >
              <select
                onChange={[Function]}
                value="currentlyReading"
              >
                <option
                  disabled={true}
                  value="move"
                >
                  Move to...
                </option>
                <option
                  defaultValue={true}
                  value="currentlyReading"
                >
                  Currently Reading
                </option>
                <option
                  defaultValue={false}
                  value="wantToRead"
                >
                  Want to Read
                </option>
                <option
                  defaultValue={false}
                  value="read"
                >
                  Read
                </option>
                <option
                  defaultValue={false}
                  value="none"
                >
                  None
                </option>
              </select>
            </div>
          </div>
          <div
            className="book-title"
          >
            The Linux Command Line
          </div>
          <div
            className="book-authors"
          >
            William E. Shotts, Jr.
          </div>
        </div>
      </li>
      <li>
        <div
          className="book"
        >
          <div
            className="book-top"
          >
            <div
              className="book-shelf-changer"
            >
              <select
                onChange={[Function]}
                value="currentlyReading"
              >
                <option
                  disabled={true}
                  value="move"
                >
                  Move to...
                </option>
                <option
                  defaultValue={true}
                  value="currentlyReading"
                >
                  Currently Reading
                </option>
                <option
                  defaultValue={false}
                  value="wantToRead"
                >
                  Want to Read
                </option>
                <option
                  defaultValue={false}
                  value="read"
                >
                  Read
                </option>
                <option
                  defaultValue={false}
                  value="none"
                >
                  None
                </option>
              </select>
            </div>
          </div>
          <div
            className="book-title"
          >
            The Linux
          </div>
          <div
            className="book-authors"
          >
            Shotts Jr.
          </div>
        </div>
      </li>
    </ol>
  </div>
</div>
`;
